### ISSUE TITLE
data_loading & online  valid dice

### ISSUE THREAD

**Issue by jk-blacker** [MIC-DKFZ/nnUNet#406]
Hi,Fabian,i  have some  problem as follows:

1.  I'm confused about the meaning of **self.num_batch_per_epoch** ?           Can I understand like this： batch_size *num_batch_per_epoch =Total amount of data set       ？？？？

2.  
2020-11-26 20:03:59.203893: train loss : 0.0590
2020-11-26 20:04:02.633565: validation loss: -0.1154
2020-11-26 20:04:02.644315: Average global foreground Dice: [0.28672218950674405]
2020-11-26 20:04:02.645567: (interpret this as an estimate for the Dice of the different classes. This is not exact.)
2020-11-26 20:04:03.289192: lr: 0.009955
2020-11-26 20:04:03.289349: saving scheduled checkpoint file...
And in these lines, is the 'Average global foreground Dice: [0.28672218950674405]' the really exact valid dice？
If so，how to get the exact valid dice？

**Comment by FabianIsensee** [comment]
> progress.png: A plot of the training (blue) and validation (red) loss during training. Also shows an approximation of the evlauation metric (green). This approximation is the average Dice score of the foreground classes. It should, however, only to be taken with a grain of salt because it is computed on randomly drawn patches from the validation data at the end of each epoch, and the aggregation of TP, FP and FN for the Dice computation treats the patches as if they all originate from the same volume ('global Dice'; we do not compute a Dice for each validation case and then average over all cases but pretend that there is only one validation case from which we sample patches). The reason for this is that the 'global Dice' is easy to compute during training and is still quite useful to evaluate whether a model is training at all or not. A proper validation is run at the end of the training.

(documentation)

**Comment by FabianIsensee** [comment]
1 epoch = 250 (self.num_batch_per_epoch) iterations with whatever batch size the model has. epochs is an ill defined term in patch based training

**Comment by jk-blacker** [comment]
thank you ,i got it
